src/Cubotino_T_display.py

```diff
@@ -0,0 +1,113 @@
+        
+        font1 = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", fs1)  # font and size for first text row
+        font2 = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", fs2)  # font and size for second text row
+        disp_img = Image.new('RGB', (self.disp_w, self.disp_h), color=(0, 0, 0)) 
+        disp_draw = ImageDraw.Draw(disp_img)
+        disp_draw.text((x1, y1), r1, font=font1, fill=(255, 255, 255))    # first text row start coordinate, text, font, white color
+        disp_draw.text((x2, y2), r2, font=font2, fill=(255, 255, 255))    # second text row start coordinate, text, font, white color
+        self.disp.display(disp_img)                                       # image is plot to the display
+
+    def display_progress_bar(self, percent):
+        """ Function to print a progress bar on the display."""
+
+        # percent value printed as text 
+        fs = 40                 # font size
+        font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", fs)     # font and its size
+        text_x = int(self.disp_w/2 - (fs*len(str(percent))+1)/2)                   # x coordinate for the text starting location         
+        text_y = 15                                                           # y coordinate for the text starting location
+        disp_img = Image.new('RGB', (self.disp_w, self.disp_h), color=(0, 0, 0)) 
+        disp_draw = ImageDraw.Draw(disp_img)
+        disp_draw.text((text_x, text_y), str(percent)+'%', font=font, fill=(255, 255, 255))    # text with percent value
+        
+        # percent value printed as progress bar filling 
+        x = 15                  # x coordinate for the bar starting location
+        y = 60                  # y coordinate for the bar starting location
+        gap = 3                 # gap in pixels between the outer border and inner filling (even value is preferable) 
+        barWidth = 35           # width of the bar, in pixels
+        barLength = 135         # lenght of the bar, in pixels
+        filledPixels = int( x+gap +(barLength-2*gap)*percent/100)  # bar filling length, as function of the percent
+        disp_draw.rectangle((x, y, x+barLength, y+barWidth), outline="white", fill=(0,0,0))     # outer bar border
+        disp_draw.rectangle((x+gap, y+gap, filledPixels-1 , y+barWidth-gap), fill=(255,255,255)) # bar filling
+        
+        self.disp.display(disp_img) # image is plotted to the display
+
+
+    def show_cubotino(self):
+        """ Shows the Cubotino logo on the display."""
+                
+        image = Image.open("Cubotino_T_Logo_265x212_BW.jpg")  # opens the CUBOTino logo image (jpg file, converted and saved from pdf file)
+        image = image.resize((self.disp_w, self.disp_h))                # resizes the image to match the display.
+        self.disp.display(image)                                   # draws the image on the display hardware.
+        
```

Comment:
def test(self):
        """ Test showing some text into some rectangles."""
        
        w = self.disp_w                                            # display width, retrieved by display setting
        h = self.disp_h                                            # display height, retrieved by display setting
        disp_img = Image.new('RGB', (w, h), color=(0, 0, 0))       
        disp_draw = ImageDraw.Draw(disp_img)
        disp_draw.rectangle((2, 2, w-4, h-4), outline="white", fill=(0,0,0))    # border 1
        disp_draw.rectangle((5, 5, w-7, h-7), outline="white", fill=(0,0,0))    # border 2
        disp_draw.rectangle((8, 8, w-10, h-10), outline="white", fill=(0,0,0))  # border 3
        font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", 23)  # font
        disp_draw = ImageDraw.Draw(disp_img)
        disp_draw.text((20, 25), 'DISPLAY', font=font, fill=(255, 255, 255))    # first row text test
        disp_draw.text((23, 65), 'TEST', font=font, fill=(255, 255, 255))       # second row text test
        self.disp.display(disp_img)                                             # image is plotted to the display